docs/editors/new-editor.md

```diff
@@ -269,5 +274,95 @@ focus back to `UserTest.scala` that depends on APIs defined in `User.scala`.
 
 _Notification_:
 
-- method: `metals/didFocus`
+- method: `metals/didFocusTextDocument`
 - params: `string`, the URI of the document where the focused was moved to.
+
+## Metals server properties
+
+The Metals language server is configured through JVM system properties.
+Configuration options are added to allow customization for particular clients
+since editors interpret LSP functionality slightly differently. For example, in
+Vim the `window/logMessage` notification is always visible in the UI while in VS
+Code logs are hidden by default.
+
+### `-Dmetals.file-watcher`
+
+Possible values:
+
+- `auto` (default): use editor file watcher if supported, otherwise use Metals
+  built-in file watcher.
```

Comment:
I would add that it relies on the client supporting dynamic capabilities registration.